src/main/scala/za/co/absa/commons/spark/SchemaUtils.scala

```diff
@@ -200,4 +210,25 @@ object SchemaUtils {
 
     diff.map(_.stripPrefix("."))
   }
+
+  /**
+   * Checks if the originalSchema is a subset of subsetSchema.
+   *
+   * @param subsetSchema The schema that needs to be extracted
+   * @param originalSchema The schema that needs to have at least all t
+   * @return true if provided schemas are the same ignoring nullability
+   */
+  def isSubset(subsetSchema: StructType, originalSchema: StructType): Boolean = {
+    val fields1 = getMapOfFields(subsetSchema)
+    val fields2 = getMapOfFields(originalSchema)
+
+    fields1.values.foldLeft(true)((stillSame, field1) => {
```

Comment:
Looks correct, but a couple of things:

1. why do you call `fields1.values` when you need the lower-case name of the field (saved in the key) and then recreate it using `field1NameLc = ...`? You could use `fields1.foldLeft` to get both directly.
2. (could render point 1 moot) - this implementation seems a bit hard to read, I can imagine it would be much simpler if written in terms of:
```scala
fields1.forall { field => ... 
  // and just make sure for each subset field that a compatible field exists - by name and by type equivalency.
}
```

Reply:
Aaah, I see that @cerveada came up with the same thing a couple of minutes ahead 😆 .
This tells me that it actually might be a good idea 🆒